datalad/core/local/run.py

```diff
@@ -231,6 +237,13 @@ class Run(Interface):
             '.datalad/runinfo' directory (customizable via the
             'datalad.run.record-directory' configuration variable).""",
             constraints=EnsureNone() | EnsureBool()),
+        dry_run=Parameter(
+            # Leave out common -n short flag to avoid confusion with
+            # `containers-run [-n|--container-name]`.
+            args=("--dry-run",),
+            action="store_true",
+            doc="""Do not run the command; just display details about the
+            command execution."""),
```

Comment:
I wonder if it is worth adding a note that `glob`s in `input/output`s might not expand (at all or fully) if pointing to some paths within not yet installed subdatasets. 
I do not think we should be getting all those inputs etc,  even though a "perfect" `--dry-run` could have included listing of  (sub)datasets to install, and a number/size of files to get  - I think it would bring up its run time considerably, thus possibly making it less convenient for typical use cases, and thus not done until need is expressed.

Reply:
Good point, but I fear that the overhead to be able to state this clearly would be substantial. I would lean towards keeping it simple, for a start.

Reply:
I've added a sentence to the docstring mentioning that globs from uninstalled datasets aren't expanded.